File: Cap14CSharp/ExpressionExtender.cs
using System;
using System.Linq.Expressions;

namespace Cap14CSharp
{
    public static class ExpressionExtender
    {
        public static Boolean IsMemberAccess(this Expression expression)
        {
            if (expression == null) return false;
            return expression.NodeType == ExpressionType.MemberAccess;
        }
    }
}

File: Cap14VB/ExpressionExtender.vb
Imports System.Linq.Expressions

Public Module ExpressionExtender
  
  <System.Runtime.CompilerServices.Extension()> _
  Public Function IsMemberAccess(ByVal expression As Expression) As Boolean
    If expression Is Nothing Then
      Return False
    End If
    Return expression.NodeType = ExpressionType.MemberAccess
  End Function
End Module
